packages/engine/src/camera/systems/CameraSystem.ts

```diff
@@ -30,36 +30,58 @@ import { Object3DComponent } from '../../scene/components/Object3DComponent'
 
 const direction = new Vector3()
 const upVector = new Vector3(0, 1, 0)
-const forwardVector = new Vector3(0, 0, 1)
 const empty = new Vector3()
 const PI_2Deg = Math.PI / 180
 const mx = new Matrix4()
 const vec3 = new Vector3()
 
-const followCameraBehavior = (entity: Entity) => {
+/**
+ * Calculates and returns view vector for give angle. View vector will be at the given angle after the calculation
+ * @param viewVector Current view vector
+ * @param angle angle to which view vector will be rotated
+ * @param isDegree Whether the angle is in degree or radian
+ * @returns View vector having given angle in the world space
+ */
+export const getViewVectorFromAngle = (viewVector: Vector3, angle: number, isDegree?: boolean): Vector3 => {
```

Comment:
we don't want this to create a new Vector3 each frame, can we instead make this a function that changes the viewVector passed in directly?

Reply:
Implemented the changed suggested